{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Forecasting Conformance Profile\n",
    "\n",
    "This notebook is used to validate that an API conforms to the TROLIE specification for [Forecasting](https://trolie.energy/spec#tag/Forecasting)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1m============================= test session starts ==============================\u001b[0m\n",
      "platform linux -- Python 3.13.3, pytest-8.4.1, pluggy-1.6.0\n",
      "rootdir: /workspace\n",
      "configfile: pytest.ini\n",
      "plugins: bdd-7.3.0, env-1.1.5, anyio-4.9.0\n",
      "collected 86 items / 83 deselected / 3 selected\n",
      "\n",
      "test/forecasting/test_conformance.py \u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                 [100%]\u001b[0m\n",
      "\n",
      "=================================== FAILURES ===================================\n",
      "\u001b[31m\u001b[1m_ test_submit_a_forecast_proposal[application/vnd.trolie.rating-forecast-proposal.v1+json-data/forecast_proposal.json-application/vnd.trolie.rating-forecast-proposal-status.v1+json] _\u001b[0m\n",
      "\n",
      "fixturefunc = <function request_forecast_limits_snapshot at 0x7f8976144360>\n",
      "request = <FixtureRequest for <Function test_submit_a_forecast_proposal[application/vnd.trolie.rating-forecast-proposal.v1+json-data/forecast_proposal.json-application/vnd.trolie.rating-forecast-proposal-status.v1+json]>>\n",
      "kwargs = {'client': <test.helpers.TrolieClient object at 0x7f8975900910>}\n",
      "\n",
      "    \u001b[0m\u001b[94mdef\u001b[39;49;00m\u001b[90m \u001b[39;49;00m\u001b[92mcall_fixture_func\u001b[39;49;00m(\u001b[90m\u001b[39;49;00m\n",
      "        fixturefunc: _FixtureFunc[FixtureValue], request: FixtureRequest, kwargs\u001b[90m\u001b[39;49;00m\n",
      "    ) -> FixtureValue:\u001b[90m\u001b[39;49;00m\n",
      "        \u001b[94mif\u001b[39;49;00m inspect.isgeneratorfunction(fixturefunc):\u001b[90m\u001b[39;49;00m\n",
      "            fixturefunc = cast(Callable[..., Generator[FixtureValue]], fixturefunc)\u001b[90m\u001b[39;49;00m\n",
      "            generator = fixturefunc(**kwargs)\u001b[90m\u001b[39;49;00m\n",
      "            \u001b[94mtry\u001b[39;49;00m:\u001b[90m\u001b[39;49;00m\n",
      "                fixture_result = \u001b[96mnext\u001b[39;49;00m(generator)\u001b[90m\u001b[39;49;00m\n",
      "            \u001b[94mexcept\u001b[39;49;00m \u001b[96mStopIteration\u001b[39;49;00m:\u001b[90m\u001b[39;49;00m\n",
      "                \u001b[94mraise\u001b[39;49;00m \u001b[96mValueError\u001b[39;49;00m(\u001b[33mf\u001b[39;49;00m\u001b[33m\"\u001b[39;49;00m\u001b[33m{\u001b[39;49;00mrequest.fixturename\u001b[33m}\u001b[39;49;00m\u001b[33m did not yield a value\u001b[39;49;00m\u001b[33m\"\u001b[39;49;00m) \u001b[94mfrom\u001b[39;49;00m\u001b[90m \u001b[39;49;00m\u001b[94mNone\u001b[39;49;00m\u001b[90m\u001b[39;49;00m\n",
      "            finalizer = functools.partial(_teardown_yield_fixture, fixturefunc, generator)\u001b[90m\u001b[39;49;00m\n",
      "            request.addfinalizer(finalizer)\u001b[90m\u001b[39;49;00m\n",
      "        \u001b[94melse\u001b[39;49;00m:\u001b[90m\u001b[39;49;00m\n",
      "            fixturefunc = cast(Callable[..., FixtureValue], fixturefunc)\u001b[90m\u001b[39;49;00m\n",
      ">           fixture_result = fixturefunc(**kwargs)\u001b[90m\u001b[39;49;00m\n",
      "                             ^^^^^^^^^^^^^^^^^^^^^\u001b[90m\u001b[39;49;00m\n",
      "\n",
      "\u001b[1m\u001b[31m/home/vscode/.local/lib/python3.13/site-packages/_pytest/fixtures.py\u001b[0m:929: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "client = <test.helpers.TrolieClient object at 0x7f8975900910>\n",
      "\n",
      "    \u001b[0m\u001b[37m@then\u001b[39;49;00m(\u001b[33m\"\u001b[39;49;00m\u001b[33mthe response is 202 OK\u001b[39;49;00m\u001b[33m\"\u001b[39;49;00m)\u001b[90m\u001b[39;49;00m\n",
      "    \u001b[94mdef\u001b[39;49;00m\u001b[90m \u001b[39;49;00m\u001b[92mrequest_forecast_limits_snapshot\u001b[39;49;00m(client: TrolieClient):\u001b[90m\u001b[39;49;00m\n",
      ">       \u001b[94massert\u001b[39;49;00m client.get_status_code() == \u001b[94m202\u001b[39;49;00m\u001b[90m\u001b[39;49;00m\n",
      "\u001b[1m\u001b[31mE       assert 415 == 202\u001b[0m\n",
      "\u001b[1m\u001b[31mE        +  where 415 = get_status_code()\u001b[0m\n",
      "\u001b[1m\u001b[31mE        +    where get_status_code = <test.helpers.TrolieClient object at 0x7f8975900910>.get_status_code\u001b[0m\n",
      "\n",
      "\u001b[1m\u001b[31mtest/common/common_steps.py\u001b[0m:46: AssertionError\n",
      "\u001b[33m=============================== warnings summary ===============================\u001b[0m\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136\n",
      "  /home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136: PytestUnknownMarkWarning: Unknown pytest.mark.prism_fail - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html\n",
      "    mark = getattr(pytest.mark, tag)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136\n",
      "  /home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136: PytestUnknownMarkWarning: Unknown pytest.mark.lep - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html\n",
      "    mark = getattr(pytest.mark, tag)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136\n",
      "  /home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136: PytestUnknownMarkWarning: Unknown pytest.mark.test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html\n",
      "    mark = getattr(pytest.mark, tag)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136\n",
      "  /home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136: PytestUnknownMarkWarning: Unknown pytest.mark.forecast_snapshot - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html\n",
      "    mark = getattr(pytest.mark, tag)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136\n",
      "  /home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/plugin.py:136: PytestUnknownMarkWarning: Unknown pytest.mark.forecast_proposal - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html\n",
      "    mark = getattr(pytest.mark, tag)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_get_forecast_limits_snapshot_requires_authorization without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_get_historical_forecast_limits_snapshot_requires_authorization without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_get_regional_forecast_limits_snapshot_requires_authorization without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_updating_the_regional_forecast_limits_snapshot_requires_authorization without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_submitting_a_forecast_proposal_requires_authorization without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_obtain_forecast_proposal_status_requires_authorization without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_support_conditional_get without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_different_representations_have_different_etags without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_unknown_etag_in_conditional_request_results_in_200_ok without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_etag_changes_when_data_is_updated without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_obtaining_the_latest_forecast_snapshot without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_obtaining_the_latest_slim_forecast_snapshot without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_slim_media_type_requires_a_limit_type without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_client_should_be_permitted_to_request_applicationproblemjson_explicitly without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_client_should_be_permitted_to_accept__explicitly without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_requesting_the_slim_forecast_snapshot_requires_a_limit_type without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_bad_query_params_are_malformed_requests without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_sending_a_body_with_a_get_request_is_a_bad_request without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_media_types_are_required without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_print_snapshot_for_psr_header__false without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_get_historical_limits_forecast_snapshot without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_get_regional_limits_forecast_snapshot without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_sending_a_body_with_a_get_regional_limits_forecast_snapshot_is_a_bad_request without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_update_regional_limits_forecast_snapshot without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_get_the_forecast_proposal_status without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_submit_a_forecast_proposal without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_forecast_limits_with_offsetperiodstart without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_what_to_do_when_offsetperiodstart_is_in_the_past____query_forecast_limits_with_offsetperiodstart_that_are_in_the_past without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_forecast_limits_with_periodend without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_forecast_limits_with_monitoringset_filter without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_forecast_limits_with_resourceid_filter without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_forecast_limits_with_staticonly without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_historical_limits_forecast_snapshots_with_offsetperiodstart without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_historical_limits_forecast_snapshots_with_periodend without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_historical_limits_forecast_snapshots_with_monitoringset without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_historical_limits_forcasting_snapshots_with_resourceid without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_historical_limits_forcasting_snapshots_with_staticonly without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_regional_limits_forecast_snapshots_with_offsetperiodstart without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_regional_limits_forecast_snapshots_with_periodend without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_regional_limits_forecast_snapshots_with_monitoringset without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_regional_limits_forecast_snapshots_with_resourceid without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/forecasting/test_conformance.py::test_query_regional_limits_forecast_snapshots_with_staticonly without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268\n",
      "  ../home/vscode/.local/lib/python3.13/site-packages/pytest_bdd/scenario.py:268: PytestWarning: usefixtures() in test/seasonal/test_seasonal_conformance.py::test_obtaining_the_latest_seasonal_snapshot without arguments has no effect\n",
      "    @pytest.mark.usefixtures(*func_args)\n",
      "\n",
      "-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html\n",
      "\n",
      "\u001b[1m==================== ���� Test Results Summary ====================\n",
      "\u001b[0m\u001b[1m  ✅ Passed:      \u001b[0m\u001b[32m\u001b[1m2\n",
      "\u001b[0m\u001b[1m  �� Failed:      \u001b[0m\u001b[31m\u001b[1m1\n",
      "\u001b[0m\u001b[1m  ⚠️ Skipped:     \u001b[0m\u001b[33m\u001b[1m0\n",
      "\u001b[0m\u001b[1m  🚫 Deselected:  \u001b[0m\u001b[33m83\n",
      "\u001b[0m\u001b[1m  ⚠️ Warnings:    \u001b[0m\u001b[33m48\n",
      "\u001b[0m\u001b[32m\u001b[1m\n",
      "  ✅ Passed Tests:\n",
      "\u001b[0m\u001b[32m    - test/forecasting/test_conformance.py::test_print_snapshot_for_psr_header__false[application/vnd.trolie.forecast-limits-snapshot.v1+json]\n",
      "\u001b[0m\u001b[32m    - test/forecasting/test_conformance.py::test_print_snapshot_for_psr_header__false[application/vnd.trolie.forecast-limits-snapshot.v1+json; include-psr-header=false]\n",
      "\u001b[0m\u001b[31m\u001b[1m\n",
      "  ❌ Failed Tests:\n",
      "\u001b[0m\u001b[31m    - test/forecasting/test_conformance.py::test_submit_a_forecast_proposal[application/vnd.trolie.rating-forecast-proposal.v1+json-data/forecast_proposal.json-application/vnd.trolie.rating-forecast-proposal-status.v1+json]\n",
      "\u001b[0m\u001b[1m===============================================================\n",
      "\u001b[0m\u001b[36m\u001b[1m=========================== short test summary info ============================\u001b[0m\n",
      "\u001b[31mFAILED\u001b[0m test/forecasting/test_conformance.py::\u001b[1mtest_submit_a_forecast_proposal[application/vnd.trolie.rating-forecast-proposal.v1+json-data/forecast_proposal.json-application/vnd.trolie.rating-forecast-proposal-status.v1+json]\u001b[0m - assert 415 == 202\n",
      "\u001b[31m=========== \u001b[31m\u001b[1m1 failed\u001b[0m, \u001b[32m2 passed\u001b[0m, \u001b[33m83 deselected\u001b[0m, \u001b[33m48 warnings\u001b[0m\u001b[31m in 2.17s\u001b[0m\u001b[31m ============\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# define the URL of the API under test\n",
    "# export TROLIE_BASE_URL='http://localhost:4010'\n",
    "# export TROLIE_BASE_URL='http://host.docker.internal:8080'\n",
    "\n",
    " \n",
    "# set the timezone of the TROLIE *client*\n",
    "# https://en.wikipedia.org/wiki/List_of_tz_database_time_zones\n",
    "export TZ='America/Los_Angeles'\n",
    "#pytest --rootdir=/workspace --disable-warnings -q  --tb=line -m \"forecasting and not requires_model and (forecast_snapshot)\" \n",
    "\n",
    "lep=1\n",
    "\n",
    "\n",
    "if [ \"$lep\" -eq 1 ]; then\n",
    "    export TROLIE_BASE_URL='http://host.docker.internal:8080'\n",
    "    pytest --rootdir=/workspace --tb=auto -m \"forecasting and not requires_model\" || true\n",
    "else\n",
    "    export TROLIE_BASE_URL='http://localhost:4010'\n",
    "    pytest --rootdir=/workspace --disable-warnings -q --tb=line -m \"forecasting and not requires_model and (forecast_snapshot)\" || true\n",
    "fi\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
